### DISCUSSION TITLE
Parallel execution of Cypress Cucumber scenarios

### DISCUSSION THREAD

**Discussion by jradom** [sorry-cypress/sorry-cypress#315] [Q&A]
- [ ] I have searched issues for similar feature requests

## What is this feature about (1 sentence)
We are using Cypress cucumber preprocessor and having only a few dozen scenarios take a lot of time
## Why is it needed? What is the value? For whom do we build it?
Is it a way to make the above toll work and run scenarios in parallel?

**Comment by agoldis** [comment]
@jradom it should be easily possible - sorry-cypress doesn't care what types of tests are you running.


You can easily test by using the demo setup:
```sh
npm install -g cy2
CYPRESS_API_URL="https://sorry-cypress-demo-director.herokuapp.com/" cy2 run <your usual cypress arguments e.g. --record --parallel --key any>
```

**Reply by jradom** [reply]
No, you can't do it with scenarios because Cypress cucumber preprocessor use its own runner with Gherkin tags

**Reply by agoldis** [reply]
I never tried running cucumber tests with cypress... In theory, if it runs cypress under the hood and provides multiple specs as an input, then [changing cypress API server](https://docs.sorry-cypress.dev/cypress-agent/configuring-cypress-agent) configuration and specifying flags `--parallel --record --key anything` should enable parallelization with sorry-cypress.

**Reply by gabraham-1** [reply]
Came across the same requirement. Wanted to run tagged cucumber scenarios in parallel and was wondering it there was a way to combine the tagged command - `./node_modules/.bin/cypress-tags run -e TAGS='@smoke_tests'` 
with the sorry-cypress parallel runs `npx cy2  --parallel --record --key somekey --ci-build-id local_run1`
Any guidance would be really helpful

**Comment by agoldis** [comment]
`./node_modules/.bin/cypress-tags` directly invokes `cypress` executable, using `cy2` would require changing the script. 
You can manually patch cypress (presumably on CI) before invoking `./node_modules/.bin/cypress-tags`:

- https://docs.sorry-cypress.dev/cypress-agent/configuring-cypress-agent
- run `CYPRESS_API_URL=... cy2 --version` will patch cypress
- create a script that would patch `cypress` with before running the rest, see https://github.com/sorry-cypress/cy2#examples
